src/init.cpp

```diff
@@ -118,6 +118,8 @@ using node::nPruneTarget;
 static const bool DEFAULT_PROXYRANDOMIZE = true;
 static const bool DEFAULT_REST_ENABLE = false;
 
+static constexpr size_t MIB_BYTES = 1024 * 1024;
```

Comment:
This will force all places where this is used to uint64_t, even if they are supposed to be signed integers. Not sure, but maybe a `#define` or `int` type is better, so that the sign of the surrounding code is chosen?

Reply:
Ouch, good point. I'm starting to doubt this change now.

Reply:
Thanks @MarcoFalke, this was indeed surprising to me:
> Otherwise, if the unsigned operand's conversion rank is greater or equal to the conversion rank of the signed operand, the signed operand is converted to the unsigned operand's type.https://en.cppreference.com/w/cpp/language/operator_arithmetic#Conversions

Reply:
Though I maintain `uint64_t` for `GB_BYTES` to maintain protections against overflow in `PruneGBtoMiB`.

Reply:
In an effort to validate these concerns, I ran build with `-Wsign-conversion` and `-Wsign-compare`, with the new and old code, and did not see any warnings relative to `MIB_BYTES`, `MIN_PRUNE_TARGET_GB`, or `_FOR_BLOCK_FILES`. I also reviewed each impacted line and did not identify such issues.

Prior, `uint64_t` code here: https://github.com/Empact/bitcoin/commits/2022-06-mib-bytes-sign-conversion-test
Current, `int32_t` code here: https://github.com/Empact/bitcoin/commits/2022-06-mib-bytes-sign-conversion